lib/engine/game/g_1854/step/buy_train.rb

```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
+
+require_relative '../../../step/buy_train'
+
+module Engine
+  module Game
+    module G1854
+      module Step
+        class BuyTrain < Engine::Step::BuyTrain
+          def can_entity_buy_train?(entity)
+            return true if entity.corporation? || entity.minor?
+
+            super
+          end
+
+          def buyable_trains(entity)
+            # allow purchase of 3 train when 2s are all sold
+            trains_to_buy = super
+
+            any_twos_upcoming = @game.depot.upcoming.any? { |t| t.name == '2' }
+            any_threes_upcoming = @game.depot.upcoming.any? { |t| t.name == '3' }
+            depot_three_available = trains_to_buy.any? { |t| t.name == '3' && t.owner == @game.depot }
+
+            if !any_twos_upcoming && any_threes_upcoming && !depot_three_available
+              trains_to_buy << @game.depot.upcoming.find { |t| t.name == '3' }
```

Comment:
This is for the 2+ and 3+ trains.  Those are in their own separate stack that is partially independent, so this makes sure those are available at the right times.

Reply:
This section of the method (lines 20–26) doesn't have any references to 1+, 2+ or 3+ trains, just 2 and 3 trains.
I don't follow the logic here. You're checking that there are no 2-trains in the `depot.upcoming` queue, but there is at least one 3-train in it. Then you check whether `trains_to_buy` includes a 3-train owned by the depot. Won't all the trains in `trains_to_buy` be owned by the depot?

Reply:
I think I see what's going on now.  This was a carryover from when I had the trains organized differently:
https://github.com/andrewzwicky/18xx/commit/672939fddf9601278eefd86601b115cfe5d1810b

You're right, this could likely be removed now.  I'll do a quick test to make sure things still work as expected.